benches/bandwidth.rs

```diff
@@ -0,0 +1,176 @@
+use std::convert::TryInto;
+
+use criterion::measurement::WallTime;
+use criterion::{black_box, criterion_group, criterion_main, BenchmarkGroup, Criterion};
+use rand::distributions::{Distribution, Standard};
+use rand::rngs::SmallRng;
+use rand::SeedableRng;
+use reed_solomon_erasure::galois_8::ReedSolomon;
+
+type Shards = Vec<Vec<u8>>;
+
+fn create_shards(block_size: usize, data: usize, parity: usize) -> Shards {
+    let mut small_rng = SmallRng::from_entropy();
+
+    let mut shards = Vec::new();
+
+    // Create data shards with random data
+    shards.resize_with(data, || {
+        Standard
+            .sample_iter(&mut small_rng)
+            .take(block_size)
+            .collect()
+    });
+
+    // Create empty parity shards
+    shards.resize_with(data + parity, || {
+        let mut vec = Vec::with_capacity(block_size);
+        vec.resize(block_size, 0);
+        vec
+    });
+
+    shards
+}
+
+fn rs_encode_benchmark(
+    group: &mut BenchmarkGroup<WallTime>,
+    block_size: usize,
+    data_shards: usize,
+    parity_shards: usize,
+) {
+    let size = block_size * data_shards;
+
+    group.throughput(criterion::Throughput::Bytes(size.try_into().unwrap()));
+
+    group.bench_function(format!("{}+{}", data_shards, parity_shards), |b| {
+        let mut shards = create_shards(block_size, data_shards, parity_shards);
+        let rs = ReedSolomon::new(data_shards, parity_shards).unwrap(); // 3 data shards, 2 parity shards
+
+        assert_eq!(shards.len(), data_shards + parity_shards);
+        assert_eq!(shards.last().unwrap().len(), block_size);
+
+        b.iter(|| {
+            rs.encode(black_box(&mut shards)).unwrap();
+        });
+    });
+}
+
+fn rs_reconstruct_benchmark(
+    group: &mut BenchmarkGroup<WallTime>,
+    block_size: usize,
+    data_shards: usize,
+    parity_shards: usize,
+    num: usize,
+) {
+    let size = block_size * data_shards;
+
+    group.throughput(criterion::Throughput::Bytes(size.try_into().unwrap()));
+
+    group.bench_function(format!("{}+{}", data_shards, parity_shards), |b| {
+        let mut shards = create_shards(block_size, data_shards, parity_shards);
+        let rs = ReedSolomon::new(data_shards, parity_shards).unwrap(); // 3 data shards, 2 parity shards
+
+        assert_eq!(shards.len(), data_shards + parity_shards);
+        assert_eq!(shards.last().unwrap().len(), block_size);
+
+        // Construct the parity shards
+        rs.encode(&mut shards).unwrap();
+
+        let mut calculated: Vec<Option<Vec<u8>>> = shards.into_iter().map(Some).collect();
+
+        b.iter(|| {
+            (0..num).for_each(|i| calculated[i] = None);
+            rs.reconstruct(black_box(&mut calculated)).unwrap();
+        });
+    });
+}
+
+fn encode(c: &mut Criterion) {
+    // let plot_config = PlotConfiguration::default();
```

Comment:
Are these comments necessary?

Reply:
They are not.

I left them in because the html report doesn't provide the most useful graphs. You could perhaps change settings using this to create more useful graphs. I'll remove these comments for now though.